paasta_tools/utils.py

```diff
@@ -1340,3 +1340,16 @@ def format_table(rows, min_spacing=2):
             expanded_rows.append(expanded_row)
 
     return [(' ' * min_spacing).join(r) for r in expanded_rows]
+
+
+def deep_merge_dictionaries(source, destination):
```

Comment:
@Rob-Johnson either an `AttributeError` or a `KeyError` will be raised. Most other implementations I've seen just check for this and raise an exception. I suppose in this case we'll want to default to the per-instance configs.